study-builder/tenants/testboston/rules/link-admin-accounts.js

```diff
@@ -0,0 +1,119 @@
+  const isAdminClient = context.clientMetadata.isAdminClient === 'true';
+
+  if (!isAdminClient || context.connection !== googleConnection) {
+    console.log('Either not admin client or not google connection, skipping account linking');
+    return callback(null, user, context);
+  }
+
+  const adminIdentity = user.identities.find(identity => identity.connection === adminConnection);
+  if (adminIdentity) {
+    console.log('Current user already has admin identity, meaning google identity has already been linked');
+    return callback(null, user, context);
+  }
+
+  request({
+    url: auth0.baseUrl + '/users',
+    headers: {
+      Authorization: 'Bearer ' + auth0.accessToken
+    },
+    qs: {
+      search_engine: 'v3',
+      q: `identities.connection:"${adminConnection}" AND email:"${user.email}"`
+    }
+  }, function(err, response, body) {
+    if (err) {
+      console.log('Error while searching for admin user with email:', user.email);
+      console.log(err);
+      return callback(err);
+    }
+
+    if (response.statusCode !== 200) {
+      console.log('Failed to search for admin user with email:', user.email);
+      console.log(body);
+      return callback(new Error(body));
+    }
+
+    const data = JSON.parse(body);
+    if (data.length !== 1) {
+      console.log('Expected to find one admin user but found', data.length);
+      return callback(new Error("Could not find a corresponding study admin account"));
+    }
+
+    const adminUser = data[0];
+    const provider = user.identities[0].provider;
+    const providerUserId = user.identities[0].user_id;
+
+    console.log('Current user:', user);
+    console.log('Found admin user:', adminUser);
+
+    const mergeCustomizer = function(objectValue, sourceValue) {
+      if (_.isArray(objectValue)) {
+        return sourceValue.concat(objectValue);
+      }
+    };
+    const mergedUserMetadata = _.merge({}, user.user_metadata, adminUser.user_metadata, mergeCustomizer);
+    const mergedAppMetadata = _.merge({}, user.app_metadata, adminUser.app_metadata, mergeCustomizer);
+
+    console.log('Merged user_metadata:', mergedUserMetadata);
+    console.log('Merged app_metadata:', mergedAppMetadata);
+
+    auth0.users.updateUserMetadata(adminUser.user_id, mergedUserMetadata)
+      .then(auth0.users.updateAppMetadata(adminUser.user_id, mergedAppMetadata))
+      .then(function() {
+        request.post({
+          url: auth0.baseUrl + '/users/' + adminUser.user_id + '/identities',
+          headers: {
+            Authorization: 'Bearer ' + auth0.accessToken
+          },
+          json: {
+            provider: provider,
+            user_id: String(providerUserId)
+          }
+        }, function(err, response, body) {
+          if (err) {
+            console.log('Error while linking identities');
+            console.log(err);
+            return callback(err);
+          }
+
+          if (response.statusCode >= 400) {
```

Comment:
It looks like a `201` is the status code for a successful request to this endpoint, so a `!== 201` would net us a bit more safety against false positives here.
```suggestion
          if (response.statusCode !== 201) {
```

Reply:
Yup!